packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -0,0 +1,253 @@
+import errorHandler from "@nhs/fhir-middy-error-handler"
+import {v4 as uuidv4} from "uuid"
+
+const logger = new Logger({serviceName: "updatePrescriptionStatus"})
+const client = new DynamoDBClient({region: "eu-west-2"})
+const tableName = process.env.TABLE_NAME
+
+interface DynamoDBItem {
+  RequestID: string | undefined;
+  PrescriptionID: string;
+  PatientNHSNumber: string;
+  PharmacyODSCode: string;
+  TaskID: string;
+  LineItemID: string;
+  TerminalStatusIndicator: string;
+  LastModified: string;
+  Timestamp: string;
+  RequestMessage: any;
+}
+
+const lambdaHandler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const xRequestId = event.headers["x-request-id"]
+    let requestBody
+    try {
+      requestBody = JSON.parse(event.body || "")
+    } catch (jsonParseError) {
+      logger.error("Error parsing JSON", {jsonParseError: jsonParseError})
+      const errorResponseBody = {
+        resourceType: "OperationOutcome",
+        issue: [
+          {
+            code: "value",
+            severity: "error",
+            details: {
+              coding: [
+                {
+                  system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
+                  code: "INVALID_JSON",
+                  display: "Invalid JSON format in request body"
+                }
+              ]
+            }
+          }
+        ]
+      }
+      return {
+        statusCode: 400,
+        body: JSON.stringify(errorResponseBody),
+        headers: {
+          "Content-Type": "application/fhir+json",
+          "Cache-Control": "no-cache"
+        }
+      }
+    }
+
+    const entries = requestBody.entry
+
+    if (!entries || entries.length === 0) {
+      logger.error("Missing required fields")
+      const errorResponseBody = {
+        resourceType: "OperationOutcome",
+        issue: [
+          {
+            code: "value",
+            severity: "error",
+            details: {
+              coding: [
+                {
+                  system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
+                  code: "MISSING_VALUE",
+                  display: "Missing required fields"
+                }
+              ]
+            }
+          }
+        ]
+      }
+      return {
+        statusCode: 400,
+        body: JSON.stringify(errorResponseBody),
+        headers: {
+          "Content-Type": "application/fhir+json",
+          "Cache-Control": "no-cache"
+        }
+      }
+    }
+
+    const responseBundle: any = {
+      resourceType: "Bundle",
+      id: uuidv4(),
+      meta: {
+        lastUpdated: new Date().toISOString()
+      },
+      type: "transaction-response",
+      entry: []
+    }
+
+    for (const entry of entries) {
+      logger.info("Processing entry", {entry: entry})
+
+      const entry_resource = entry.resource
+      logger.info("Processed the entry resource", {processedEntryResource: entry_resource})
```

Comment:
Can probably drop this log as its a little misleading, nothing has actually been processed since the last log line, we've just pulled some data out of a field.